src/blockdev.rs

```diff
@@ -930,13 +930,13 @@ pub fn find_efi_vendor_dir(efi_mount: &Mount) -> Result<PathBuf> {
     let mut vendor_dir: Vec<PathBuf> = Vec::new();
     for ent in p.read_dir()? {
         let ent = ent.with_context(|| format!("reading directory entry in {}", p.display()))?;
-        if ent.file_name() == "BOOT" {
-            continue;
-        }
         if !ent.file_type()?.is_dir() {
             continue;
         }
-        vendor_dir.push(ent.path());
+        let path = ent.path();
+        if path.join("grub.cfg").is_file() {
```

Comment:
I'm a little uncertain about us saying we'll always use grub.  I guess it's OK for now, if we ever do something else we can start looking for sd-boot or whatever too.

Reply:
Yeah, I wasn't sure if there was a better heuristic. Doing this somewhat makes sense though because this code is concerned with adding the `bootuuid.cfg` dropins which are currently sourced by `grub.cfg`.